auto_round/autoround.py

```diff
@@ -876,10 +884,16 @@ def quant_block(self, block, input_ids, input_others, q_input=None, device=torch
                 current_output = move_input_to_device(current_output, device)
 
                 output_q = block_forward(
-                    block, current_input_ids, current_input_others, self.amp, self.amp_dtype, device
+                    block,
+                    current_input_ids,
+                    current_input_others,
+                    self.amp,
+                    self.amp_dtype,
+                    self.amp_device_type,
+                    device,
                 )
                 if self.amp and not check_is_cpu(device):
-                    with autocast(device_type="cuda", dtype=self.amp_dtype):
+                    with autocast(device_type=self.amp_device_type, dtype=self.amp_dtype):
```

Comment:
a better way is just detect whether ":" in device string, and truncate the string, as there may more device like xpu, npu, etc

Reply:
As I know, some code do not check npu or xpu device and we don't validate such devices. If we want to adapt other device, we need validate it and revise code.